src/multi-selector.js

```diff
@@ -77,6 +78,15 @@ const MultiSelector = React.createClass({
 		})
 	},
 
+	shouldComponentUpdate: function(nextProps, nextState) {
+		return nextState.searchValue !== this.state.searchValue
+			|| nextProps.initialSelectedItems !== this.state.selectedItems
+			|| nextProps.items !== this.props.items
+			|| nextState.dialogDisplayed !== this.state.dialogDisplayed
+			|| nextState.activeIndex !== this.state.activeIndex
+			|| nextState.mouseIndex !== this.state.mouseIndex;
```

Comment:
idk if this is a dumb suggestion, but could you just do `nextState !== this.state && nextProps !== this.props`?